libazureinit/src/http.rs

```diff
@@ -125,7 +123,7 @@ async fn request(
 
                         match response.error_for_status() {
                             Ok(response) => {
-                                if statuscode == StatusCode::OK {
+                                if statuscode == StatusCode::OK || statuscode == StatusCode::CREATED {
```

Comment:
i think we can remove this if statement altogether.  We're in this Ok() block so any status code we get should be safe to consider OK (2xx)